src/bin/pg_autoctl/cli_common.c

```diff
@@ -406,16 +407,13 @@ cli_create_node_getopts(int argc, char **argv,
 	 */
 	if (IS_EMPTY_STRING_BUFFER(LocalOptionConfig.pgSetup.pgdata))
 	{
-		char *pgdata = getenv("PGDATA");
-
-		if (pgdata == NULL)
+		int pgdatalen = get_env_variable("PGDATA", LocalOptionConfig.pgSetup.pgdata, MAXPGPATH);
+		if (pgdatalen <= 0 || pgdatalen >= MAXPGPATH)
```

Comment:
This is a behavioral change, since `PGDATA=""` will now fail (maybe this is fine)

Reply:
If we create a function such as `getPGDATA(char *pgdata, int size, bool error_if_not_found)` we should then probably test that we can actually use the pgdata that was found. Empty value does not make sense for pgdata, we need to be able to `mkdir -p` the directory, but also we need the directory to NOT exists yet when we do `initdb`. We could remove the directory before `initdb`, that said.

Reply:
I think it makes sense to do these checks, but more in a separate PR. Always nice to keep refactor changes and behavioural changes mostly separate.